R/checkPythonDeps.R

```diff
@@ -0,0 +1,350 @@
+  return(list(
+    major = as.integer(major),
+    minor = as.integer(minor),
+    patch = as.integer(patch),
+    releaseType = as.character(releaseType),
+    releaseVersion = as.integer(releaseVersion)
+  ))
+}
+
+#' Print Python version string
+#'
+#' This function creates a string representation of a Python version.
+#'
+#' @param pythonVerion A named list representing the Python version.
+#' @return A string representation of the Python version.
+printPythonVersion <- function(pythonVerion) {
+  return(paste0(
+    ifelse(is.na(pythonVerion$major), "", as.character(pythonVerion$major)),
+    ifelse(is.na(pythonVerion$minor), "", paste0(".", as.character(pythonVerion$minor))),
+    ifelse(is.na(pythonVerion$patch), "", paste0(".", as.character(pythonVerion$patch))),
+    ifelse(is.na(pythonVerion$releaseType), "", pythonVerion$releaseType),
+    ifelse(is.na(pythonVerion$releaseVersion), "", pythonVerion$releaseVersion)
+  ))
+}
+
+#' Extract Python Version
+#'
+#' This function extracts a named list representing Python version from a string by matching a regular expression.
+#'
+#' @param versionString A Python version string like "3.10.0a0"
+#' @return A named list representing the Python version
+extractPythonVersion <- function(versionString) {
+  # Check PIP 440 and implementation for an even more extensive version regex. This represents a resonable subset
+  # of what to expect in the wild (https://www.python.org/dev/peps/pep-0440/#version-scheme), but might need to be
+  # extended for specific use cases such as post-releases or local versions.
+  versionRegex <- regex("
+    v?                    # Leading v
+    (?<major>[0-9]+)\\.?  # Major version
+    (?<minor>[0-9]+)?\\.? # Minor version
+    (?<patch>[0-9]+)?     # Patch version
+    [-_\\.]?              # Possible separator
+    (?<rtype>alpha|a|beta|b|preview|pre|c|rc)? # Release type
+    [-_\\.]?             # Possible separator
+    (?<rnum>[0-9]+)?     # Release number
+    ", comments = TRUE)
+  # str_match a matrix with first column containing the entire match and then all individual group patterns with
+  # columns named after the group names
+  matches <- str_match(versionString, versionRegex)
+  return(createPythonVersion(
+    major = matches[, "major"],
+    minor = matches[, "minor"],
+    patch = matches[, "patch"],
+    releaseType = matches[, "rtype"],
+    releaseVersion = matches[, "rnum"]
+  ))
+}
+
+#' Compare Python Versions
+#'
+#' This function compares two Python versions
+#'
+#' @param operator Comparison operator. One of '==', '!=', '>', '>=', '<' or '<='
+#' @param first First Python version (represented as named list)
+#' @param second Second Python version (represented as named list) to compare against
+#' @param strict Strict comparison of version components (e.g. 3.10 != 3.10.0)
+#' @return TRUE operator is satisfied, FALSE otherwise
+comparePythonVersions <- function(operator, first, second, strict = TRUE) {
```

Comment:
I didn't thoroughly read the entire function, but I have the feeling R's `numeric_version` class supports all these comparison operations the way you want do?

Reply:
`numeric_version` does not do Python version strings in a consistent manner though :-/